### DISCUSSION TITLE
How to use Monday-based weeks as timeUnit?

### DISCUSSION THREAD

**Discussion by bmesuere** [vega/vega-lite#7389] [Q&A] [answered]
I'm creating a graph where I aggregate data by week. I do this by setting the `timeUnit` of a temporal field to `yearweek`. The aggregations works fine, but vega-lite (and vega?) seem to use Sunday as the first day of the week. 

Is there a way to easily use Monday-based weeks instead (as per ISO standard)? Right now I'm shifting the dates by one day in preprocessing which is a bit hackish.

**Comment by jheer** [comment, marked as answer]
Unfortunately, I think this will require extensions to Vega itself, starting with the vega-time package: https://github.com/vega/vega/tree/master/packages/vega-time

(PRs welcome!)

**Reply by bmesuere** [reply]
I can look into creating a PR next week. Is adding an `isoweek` time unit the way to go?

**Reply by jheer** [reply]
Thanks! Yeah, something like that is probably the right path. [d3-time](https://github.com/d3/d3-time) uses `timeWeek` for Sunday-based weeks, then provides `timeSunday`, `timeMonday`, `timeTuesday`, _etc._ for shifting the week start day.

I'm not sure if we would want `isoweek` or a different (perhaps more descriptive?) name. We could always revise naming before release once an implementation is in place. A more pressing question is if we want to add new varieties of week time units, or add a separate parameter to specify the week start day (e.g., `0` (default) for Sunday, `1` for Monday, following the JS `Date.getDay()` numbering), similar to how we specify UTC vs. local time.
